src/Features/CSharp/Portable/Completion/CompletionProviders/AwaitCompletionProvider.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    [ExportCompletionProvider(nameof(AwaitCompletionProvider), LanguageNames.CSharp)]
+    [ExtensionOrder(After = nameof(KeywordCompletionProvider))]
+    [Shared]
+    internal sealed class AwaitCompletionProvider : LSPCompletionProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public AwaitCompletionProvider()
+        {
+        }
+
+        public override ImmutableHashSet<char> TriggerCharacters => CompletionUtilities.CommonTriggerCharactersWithArgumentList;
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var workspace = document.Project.Solution.Workspace;
+            var syntaxContext = CSharpSyntaxContext.CreateContext(workspace, semanticModel, position, cancellationToken);
+
+            if (!syntaxContext.IsAwaitKeywordContext(position))
+            {
+                return;
+            }
+
+            var method = syntaxContext.TargetToken.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (method is not null && !method.GetModifiers().Any(SyntaxKind.AsyncKeyword))
+            {
+                var symbol = semanticModel.GetDeclaredSymbol(method, cancellationToken) as IMethodSymbol;
+                if (symbol is not null && IsTask(symbol.ReturnType))
+                {
+                    context.AddItem(CommonCompletionItem.Create("await", string.Empty, CompletionItemRules.Default, glyph: Glyph.Keyword, inlineDescription: CSharpFeaturesResources.Make_container_async));
+                }
+            }
+
+            static bool IsTask(ITypeSymbol returnType)
```

Comment:
You probably want to check more than just name, e.g.
http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs,31fc2ba43ad5b50c,references

Reply:
It makes sense not to add `async` for void returning method, but please add some comment to make the intention clearer.

Reply:
> You probably want to check more than just name, e.g.
http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs,31fc2ba43ad5b50c,references

Any good place to extract this as a helper function instead of duplicating the code?

Reply:
Maybe move `IsTaskLike` and `KnownTypes` into a separate `MakeMethodAsynchronousHelper.cs` in the same folder

Reply:
In fact, a lot of the logic can be shared between MakeMethodAsynchronousCodeFixProvider and the new completion provider. I'd suggest extract the core of `MakeMethodAsynchronousCodeFixProvider` into a language service `MakeMethodAsynchronousService`, so the completion will work in both C# and VB.